src/v/storage/segment_index.cc

```diff
@@ -177,27 +203,26 @@ ss::future<bool> segment_index::materialize_index() {
 
 ss::future<> segment_index::drop_all_data() {
     reset();
-    return _out.truncate(0);
+    co_return co_await (co_await open()).truncate(0);
```

Comment:
missing close

Reply:
nice catch. based on travis' comment above it does look like a missing close.

Reply:
This one is fixed in the later commit "fix segment_index destroy-without-close on files"